src/app/components/match/match.component.ts

```diff
@@ -35,44 +38,81 @@ export class MatchComponent implements OnInit {
+    this.volunteeringRequestsService.getUnmatched().then(requests => {
+        this.volunteeringRequests = requests; });
+    this.volunteeringOffersService.getUnmatched().then(offers => { this.volunteeringOffers = offers; });
+    this.volunteeringEventsService.getPlannedEvent().then( planned => { this.plannedEvents = planned; });
+
 
     this.date = this.calendar.getToday();
-    const now  = new Date();
-    this.time = { hour: now.getHours(), minute: now.getMinutes() };
 
     this.matchForm = this.formBuilder.group({
+      title: [undefined, Validators.required],
       request: [undefined, Validators.required],
       offer: [undefined, Validators.required],
-      date: [now, Validators.required],
+      eventDate: [this.date, Validators.required],
     });
   }
 
   onSelectRequest(request: VolunteeringRequestModel): void {
-    console.log(request);
     this.selectedRequest = request;
+    this.matchForm.get('request').setValue(this.selectedRequest._id);
+    const currentTitle = this.matchForm.get('title').value;
+    if ( currentTitle == null ) {
+      this.matchForm.get('title').setValue(this.selectedRequest.title + ' עבור ' + this.selectedRequest.organization + ' על ידי ');
+    } else {
+      this.matchForm.get('title').setValue(this.selectedRequest.title + ' עבור ' +
+       this.selectedRequest.organization + ' על ידי ' + currentTitle.split('על ידי')[1].trim());
+    }
   }
 
-  onSelectOffer(offer: VolunteeringOfferModel): void {
-    console.log(offer);
+  onSelectOffer(offer: VolunteeringOfferModel) {
     this.selectedOffer = offer;
-  }
-
-  updateEventTime() {
-    this.eventDate = new Date(this.date.year, this.date.month - 1, this.date.day, this.time.hour, this.time.minute);
+    this.matchForm.get('offer').setValue(this.selectedOffer._id);
+    const currentTitle = this.matchForm.get('title').value;
+    if ( currentTitle == null ) {
+      this.matchForm.get('title').setValue(' על ידיי ' + this.selectedOffer.organization);
+    } else {
+        this.matchForm.get('title').setValue(currentTitle.split('על ידי')[0].trim() + ' על ידי ' + this.selectedOffer.organization);
+    }
   }
 
   sendData() {
-    const event: VolunteeringEvent = this.matchForm.value;
-    console.log(event);
+    const event: any = this.matchForm.value;
+    this.volunteeringEvent = {} as any;
+    this.volunteeringEvent.title = event.title;
+    this.volunteeringEvent.offer = event.offer;
+    this.volunteeringEvent.request = event.request;
+    this.volunteeringEvent.date = new Date( event.eventDate.year, event.eventDate.month - 1,
+        event.eventDate.day );
+    this.volunteeringEventsService.create(this.volunteeringEvent).then(response => {
+        if (response) {
+            this.volunteeringRequestsService.setMatchced(event.request, true);
+            this.volunteeringOffersService.setMatchced(event.offer, true);
+            this.volunteeringRequests = this.volunteeringRequests.filter(item => item._id !== event.request);
```

Comment:
It looks like after every match, you filter the latest match only out of the list. Don't you want to simply remove it from the list?
Maybe I just don't understand what you did here...

Reply:
This is how you remove it from the list.